write_buffer/src/kafka.rs

```diff
@@ -205,18 +214,30 @@ impl WriteBufferReading for KafkaBufferConsumer {
             let database_name = self.database_name.clone();
             let trace_collector = self.trace_collector.clone();
 
+            // prepare a metric recorder for this sequencer
+            let attributes = metric::Attributes::from([("database", database_name.clone().into())]);
+            let write_buffer_ingest_entry_size =
+                self.write_buffer_ingest_entry_size.recorder(attributes);
+
             let stream = consumer
                 .stream()
                 .map(move |message| {
                     let message = message?;
 
-                    let kafka_headers = message.headers().into_iter().flat_map(|headers| (0..headers.count()).map(|idx| headers.get(idx).unwrap()));
+                    let kafka_headers = header_iter(&message);
                     let headers = IoxHeaders::from_headers(kafka_headers, trace_collector.as_ref())?;
-
                     let payload = message.payload().ok_or_else::<WriteBufferError, _>(|| {
                         "Payload missing".to_string().into()
                     })?;
 
+                    // Estimate size of data read from kafka as
+                    // payload len + key len + headers
+                    let kafka_read_size: usize = payload.len() +
+                        message.key().map(|key| key.len()).unwrap_or(0) +
```

Comment:
Haven't tested this but I think you can do. YOLO!!

```suggestion
                        message.key().map(|key| key.len()).unwrap_or_default() +
```

Reply:
In 18f5947a